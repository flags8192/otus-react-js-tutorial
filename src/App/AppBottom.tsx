import React, { FC, memo, useContext } from 'react';
import Panel from '@src/layout/Panel';
import styled from '@emotion/styled';
import store from '@src/store';

const Message = styled.p`
  margin: auto 20px;
`;

const AppBottom: FC = () => {
  const { state } = useContext(store);
  const { gameFieldPercentFilled } = state;

  return (
    <Panel role="bottomPanel">
      <Message role="messageResult">Результат:</Message>
      <Message role="messagePercentFilled">{`Процент заполнения: ${gameFieldPercentFilled}`}</Message>
    </Panel>
  );
};

export default memo(AppBottom);
