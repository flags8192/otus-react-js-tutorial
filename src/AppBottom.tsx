import React, { PureComponent } from 'react';
import styled from '@emotion/styled';

const Root = styled.div`
  height: 10vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
`;

interface IAppBottom {
  countUsers: number;
}
class AppBottom extends PureComponent<IAppBottom> {
  render() {
    return (
      <Root>
        <p role="userCount">Количество пользователей: {this.props.countUsers}</p>
      </Root>
    );
  }
}

export default AppBottom;
