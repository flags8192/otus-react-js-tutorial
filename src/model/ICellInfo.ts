enum ICellInfo {
  alive = 1,
  dead = 0,
}

export default ICellInfo;
