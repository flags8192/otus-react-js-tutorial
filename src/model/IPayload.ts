interface IPayload {
  gameFieldSize?: number;
  selected?: boolean;
  cellId?: number;
  gameLevel?: string;
}

export default IPayload;
