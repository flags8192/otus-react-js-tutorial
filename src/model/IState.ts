import ICellInfo from '@src/model/ICellInfo';
import { ISelectedCells } from '@src/modules/Cell/models';

export interface IState {
  gameFieldSize: number;
  gameFieldPercentFilled: number;
  gameFieldData: Array<ICellInfo>;
  gameLevel: string;
  selectedCells: ISelectedCells;
}
