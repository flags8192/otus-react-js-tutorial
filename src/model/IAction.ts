import IPayload from './IPayload';

export interface IAction {
  type: string;
  payload?: IPayload;
}
