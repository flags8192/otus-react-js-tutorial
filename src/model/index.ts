import { IState } from '@src/model/IState';
import { IAction } from '@src/model/IAction';

export { IAction, IState };
