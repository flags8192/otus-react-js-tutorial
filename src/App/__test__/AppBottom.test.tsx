import React from 'react';
import { render, screen } from '@testing-library/react';
import AppBottom from '@src/App/AppBottom';
import Store from '@src/store';
import { IAction } from '@src/model';

let dispatch: (action: IAction) => void;

describe('AppBottom', () => {
  beforeEach(() => {
    dispatch = jest.fn();
  });

  it('Render <AppBottom> without data', () => {
    const state = {
      gameLevel: '1',
      gameFieldSize: 3,
      gameFieldPercentFilled: 10,
      gameFieldData: [1, 0, 0, 0, 0, 0, 0, 0, 0],
      selectedCells: {},
    };
    const { asFragment } = render(
      <Store.Provider value={{ dispatch, state }}>
        <AppBottom />
      </Store.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
    const dataMessage = screen.getByText(/Результат/gi);
    expect(dataMessage).toBeInTheDocument();
    const percentMessage = screen.getByText(/Процент заполнения/gi);
    expect(percentMessage).toBeInTheDocument();
  });
});
