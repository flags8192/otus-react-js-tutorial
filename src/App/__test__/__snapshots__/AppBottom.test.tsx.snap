// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBottom Render <AppBottom> without data 1`] = `
<DocumentFragment>
  <div
    class="css-cogpxn"
    role="bottomPanel"
  >
    <p
      class="css-o9gtt2"
      role="messageResult"
    >
      Результат:
    </p>
    <p
      class="css-o9gtt2"
      role="messagePercentFilled"
    >
      Процент заполнения: 10
    </p>
  </div>
</DocumentFragment>
`;
